src/common.c

```diff
@@ -85,11 +85,46 @@ struct ofi_common_locks common_locks = {
 	.util_fabric_lock = PTHREAD_MUTEX_INITIALIZER,
 };
 
+int ofi_fork_unsafe;
 size_t ofi_universe_size = 1024;
 int ofi_av_remove_cleanup;
 char *ofi_offload_coll_prov_name = NULL;
 
 
+void ofi_params_init(void)
+{
+	fi_param_define(NULL, "fork_unsafe", FI_PARAM_BOOL,
+			"Whether use of fork() may be unsafe for some providers "
+			"(default: no). Setting this to yes could improve "
+			"performance at the expense of making fork() potentially "
+			"unsafe");
```

Comment:
Doesn't this result in the same parameter being defined multiple times?  The provider param is null here, so the core and all DL providers will define the same parameter.

I think the DL providers only need to call param_get to read the value, with the assumption that libfabric core has already defined the param.

Reply:
Good call. Fixed.